app/util/conf/master_config_loader.py

```diff
@@ -38,3 +39,9 @@ def configure_postload(self, conf):
         # where to store results on the master
         conf.set('results_directory', join(base_directory, 'results', 'master'))
         conf.set('timings_directory', join(base_directory, 'timings', 'master'))  # timing data
+
+        # Set protocol scheme (by default its set to 'http')
+        https_cert_file = Configuration['https_cert_file']
+        https_key_file = Configuration['https_key_file']
+        if https_cert_file and https_key_file:
+            conf.set('protocol_scheme', 'https')
```

Comment:
I think you could do this in the `configure_postload` section of base_config_loader. That method contains conf values derived from other conf values. It might allow you to eliminate a little bit of duplication.